File: python/leetcode/27. Remove Element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        first = 0
        last = len(nums) - 1
        lenn = 0

        while first <= last:
            if nums[last] == val:
                nums[last] = val - 1
                last -= 1
                lenn += 1
            else:
                if nums[first] == val:
                    nums[first] = nums[last]
                    first += 1
                    nums[last] = val - 1
                    last -= 1
                    lenn += 1
                else:
                    first += 1
        return len(nums) - lenn

File: go/leetcode/27. Remove Element.go
func removeElement(nums []int, val int) int {
    first := 0
    last := len(nums) - 1
    lenn := 0
    
    for first <= last {
        if nums[last] == val {
            nums[last] = val - 1
            last--
            lenn++
        } else {
            if nums[first] == val {
                nums[first] = nums[last]
                first++
                nums[last] = val - 1
                last--
                lenn++
            } else {
                first++
            }
        }
    }
    
    return len(nums) - lenn
}

  